app/models/tag.rb

```diff
@@ -20,5 +16,23 @@ def self.parse_update_actions(client_ids, server_ids)
       remove: server_ids - client_ids
     }
   end
-end
 
+  # String -> [Tag]
+  def self.search_by_names(phrase)
```

Comment:
all of the tag search logic is incredibly elegant! easy to follow and well-executed! great work! 👍